utils/databake/derive/src/lib.rs

```diff
@@ -31,6 +32,40 @@ use synstructure::{AddBounds, Structure};
 ///     pub age: u32,
 /// }
 /// ```
+///
+/// # Custom baked type
+///
+/// To bake to a different type than this, use `custom_bake`
+/// and implement `CustomBake`.
```

Comment:
I think having a trait for this is overkill if you can provide a method to the macro

Reply:
Reasons I made it a trait:

1. Gives a place to enforce the strange safety requirement in the unsafe version
2. And since we have it for unsafe, we can also use it for safe 

How do you suggest handling the safety requirement without a trait?